lib/iris/_merge.py

```diff
@@ -1319,10 +1319,10 @@ def axis_and_name(name):
                     # scalars for each source cube. The keys of an OrderedDict
                     # are used to retain the ordering of source cubes but to
                     # remove any duplicate tuples.
-                    cells = OrderedDict(
+                    cells = list(OrderedDict(
```

Comment:
It looks like the `iter` produced from `.keys()` is smart enough to handle the two `len` before the `enumerate`, so probably not.